### PR TITLE
Add setSubredditSticky to api/modTools

### PR DESCRIPTION
**Pull request by madbook**
Adds support for the endpoint needed to sticky/unsticky posts (AKA pin/unpin announcements).

👓 @scarow @birakattack

### CODE REVIEW

File: src/apis/modTools.es6.js [hunk 1]
```diff
@@ -16,4 +16,27 @@ const approve = (apiOptions, fullname) => {
...
-export default { remove, approve }
\ No newline at end of file
+/**
+ * Set or unset a stickied post (AKA an "Annoucement").
+ * See also: https://www.reddit.com/dev/api#POST_api_set_subreddit_sticky
+ * @function
+ * @param {Object} apiOptions
+ * @param {string} fullname The fullname of the target post
+ * @param {boolean} isStickied Whether to sticky or unsticky the post
+ * @param {number?} [stickyNum] Allows for specifying the "slot" to sticky the post
```
Comment (scarow) [inline src/apis/modTools.es6.js]: WOWOWOWOW what is this fanciness??

File: src/apis/modTools.es6.js [hunk 2]
```diff
@@ -16,4 +16,27 @@ const approve = (apiOptions, fullname) => {
...
+  };
+
+  if (stickyNum) {
+    body.num = stickyNum;
+  }
+
+  return apiRequest(apiOptions, 'POST', 'api/set_subreddit_sticky', { body, type: 'form' });
+}
+
+export default { remove, approve, setSubredditSticky }
```
Comment (scarow) [inline src/apis/modTools.es6.js]: Should be a newline at end of file

File: src/apis/modTools.es6.js (hunk 1 above)
Comment (scarow) [inline src/apis/modTools.es6.js]: Also I think this one is kinda messed up -- maybe due to the question mark?

`@param {number?} [stickyNum]`

**Comment by scarow** [comment]
:haircut_man:

File: src/apis/modTools.es6.js (hunk 1 above)
Comment (madbook) [inline src/apis/modTools.es6.js]: oooh good catch, i think the question mark is supposed to be prefixed, not postfixed 😱

File: src/apis/modTools.es6.js (hunk 2 above)
Comment (madbook) [inline src/apis/modTools.es6.js]: there is!

File: src/apis/modTools.es6.js (hunk 1 above)
Comment (madbook) [inline src/apis/modTools.es6.js]: the `?` indicates that it's nullable, i.e. that `stickyNum` is either a `number` or `null`, while the `[]` around the argument name indicates that it's optional

**Comment by madbook** [comment]
:haircut_man:

**Comment by scarow** [comment]
:fish:

**Comment by madbook** [comment]
7efcf0ae25934384e277d57973f8fe09cdebabc3 included in release v3.34.0

**Pull request closed without merging**